apollo-federation/src/operation/merging.rs

```diff
@@ -206,7 +206,7 @@ impl SelectionSet {
                             ));
                         };
                         fields
-                            .entry(other_key)
+                            .entry(other_key.to_owned_key())
```

Comment:
This is not easy to rewrite to borrowed keys, because it creates intermediate maps grouping selections that have the same key. We can probably rewrite merging without the intermediate maps entirely later.